framework/test_app/runners/k8s/k8s_base_runner.py

```diff
@@ -299,6 +299,34 @@ def _reuse_namespace(self) -> k8s.V1Namespace:
         logger.info("Reusing namespace: %s", self.k8s_namespace.name)
         return self.k8s_namespace.get()
 
+    def _create_pod_monitoring(self, template, **kwargs) -> None:
+        if not kwargs["namespace_name"]:
+            raise _RunnerError(
+                "namespace_name required to create PodMonitoring resource"
+            )
+        if not kwargs["deployment_id"]:
+            raise _RunnerError(
+                "deployment_id required to create PodMonitoring resource"
+            )
+        if not kwargs["pod_monitoring_name"]:
+            raise _RunnerError(
+                "pod_monitoring_name required to create PodMonitoring resource"
+            )
+        pod_monitoring = self._create_from_template(
+            template, custom_object=True, **kwargs
+        )
+        if pod_monitoring.metadata.namespace != kwargs["namespace_name"]:
+            raise _RunnerError(
+                "PodMonitoring resource created with unexpected namespace: "
+                f"{pod_monitoring.metadata.namespace}"
+            )
```

Comment:
Instead, do a check similar to the one in `_create_gamma_route`; just for `PodMonitoring` kind

```py
        if not (isinstance(route, k8s.GammaHttpRoute) and route.kind == "HTTPRoute"):
            raise _RunnerError(
                f"Expected ResourceInstance[HTTPRoute] to be created from"
                f" manifest {template}"
            )
        if route.metadata.name != kwargs["route_name"]:
            raise _RunnerError(
                "ResourceInstance[HTTPRoute] created with unexpected name: "
                f"{route.metadata.name}"
            )
```

Reply:
Used that pattern in the new commit.